### DISCUSSION TITLE
Cannot install the plugins for lazynvim.

### DISCUSSION THREAD

**Discussion by prexhu** [LazyVim/LazyVim#2651] [Q&A]
machcine: cloud_server 
OS: Ubuntu 20.04.6 LTS

I follow the installation instructions from https://www.lazyvim.org/,and when I use nvim command in my terminal, I get this error for 49/51 plugins.

Show the error `fetch failed fatal: unable to asscess 'https:///github.com/xxx.git/': gnutls_handshake() failed: The TLS connection was non-properly terminated.
**But** I manually installed the plugins and used the command `git clone ` to clone 49 to the folder `~/.local/share/nvim/lazy`

The thing that weird me is that I can use git and github in my machine for free, but the install and update command in `nvim->lazy` will show errors.

![image](https://github.com/LazyVim/LazyVim/assets/96912272/4c58af93-f69f-429f-92d9-e582304e0d8b)

And I try the command `git config http.sslVerigy false` and unset the git proxy, but it does not work for my machine.

**Comment by MartinEmrich** [comment]
Have the same experience here, trying to shoehorn neovim onto my OG Raspberry Pi 1 Model B.

Lazy tries to clone every plugin at once during initial setup (>20 for me). I guess that's just too much, probably resulting in TLS handshake timeouts for all.

But how can I limit the number of parallel clones?

**Reply by MartinEmrich** [reply]
Hmm that was easy: https://lazy.folke.io/configuration

set `concurrency = 1,`, now it clones one after another, which works for me now.

**Reply by dpetka2001** [reply]
You can gradually increase `concurrency` until you find the sweet spot. I've seen 8-10 in some other posts. Try it out.

**Reply by viscount-monty** [reply]
Same situation, trying to get nvim/lazy set up on my Model 1 B.

I can't seem to get `concurrency = 1` to work, it always tries to clone all plugins simultaneously, then fails in the same way as above.

I'm using `nvim-lua/kickstart.nvim`. In `init.lua` I tried the following:
```lua
-- NOTE: Here is where you install your plugins.
require('lazy').setup({
  -- Clone less plugins at a time for Raspberrry Pi 1 error
  checker = {
    concurrency = 1,
  },
 . . .
```
and
```lua
-- NOTE: Here is where you install your plugins.
require('lazy').setup({                                                                                 
  -- Clone less plugins at a time for Raspberrry Pi 1 error                                                                                                                                           
  concurrency = 1,                                                                                                                                       
  . . .
```

But no change in behaviour:
![image](https://github.com/user-attachments/assets/cd252938-ac34-46cf-98d3-0c5fb89ebb00)
Then eventually:
![image](https://github.com/user-attachments/assets/2367b81d-0078-4996-9c07-60f5012c2014)

**Reply by folke** [reply]
It seems that https://lazy.folke.io was built from an old branch. Just fixed that.

Check the docs on configuration for throttle.

**Reply by viscount-monty** [reply]
Thank kindly for sorting that out - all plugins installed!

**Reply by bennigraf** [reply]
Hey,
I have the exact same problem, but neither `concurrency` nor the `throttle` settings seem to have any effect. I'm adding this to `init.lua` like this:

```
require('lazy').setup({
   concurrency = 1,
   git = {
     throttle = {
       enabled = true, -- not enabled by default
       -- max 2 ops every 5 seconds
       rate = 1,
       duration = 15 * 1000, -- in ms
     },
   },
[...plugins follow here...]
```

Any ideas what I'm doing wrong here?

Best,
Benni

**Reply by theTschi** [reply]
Hey bennigraf,

for me this worked (and fixed the tls problem):

```
require('lazy').setup({
  [...plugins follow here...]
},{
   git = {
     throttle = {
       enabled = true, -- not enabled by default
       -- max 2 ops every 5 seconds
       rate = 1,
       duration = 15 * 1000, -- in ms
     },
   },
})
```

Hope this helps.
